srv/src/main/java/my/bookshop/handlers/AdminServiceHandler.java

```diff
@@ -248,6 +256,39 @@ public void addBookToOrder(AddToOrderContext context) {
 		context.setResult(updatedOrder);
 	}
 
+	@On(event = CdsService.EVENT_UPDATE)
+	public void addBooksViaCsv(CdsUpdateEventContext context, List<Csv> csvs) {
+		csvs.forEach(csv -> {
+			InputStream is = csv.getData();
+			if (is != null) {
+				try (BufferedReader br = new BufferedReader(new InputStreamReader(is))) {
+					br.lines().skip(1).forEach((line) -> {
+						String[] p = line.split(";");
+						Books book = Books.create();
+						book.setId(p[0]);
+						book.setTitle(p[1]);
+						book.setDescr(p[2]);
+						book.setAuthorId(p[3]);
+						book.setStock(Integer.valueOf(p[4]).intValue());
+						book.setPrice(BigDecimal.valueOf(Double.valueOf(p[5])));
+						book.setCurrencyCode(p[6]);
+						book.setGenreId(Integer.valueOf(p[7]));
+
+						// separate transaction per line
+						context.getCdsRuntime().changeSetContext().run(ctx -> {
+							db.run(Upsert.into(Bookshop_.BOOKS).entry(book));
+						});
+					});
+				} catch (IOException e) {
+					messages.error(MessageKeys.BOOK_IMPORT_FAILED);
+				}
+			} else {
+				db.run(Upsert.into(Bookshop_.CSV).entry(csv));
```

Comment:
We could also just ignore this, especially if we want to get rid of the unnecessary db persistence for this entity.